### PR TITLE
258 -  Support Multiplexed on Portal 

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number
#258  

## Purpose/Implementation Notes

Added support for Multiplexed downloads.

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist
- [x] Lint and unit tests pass locally with my changes

## Screenshots

N/A

### CODE REVIEW

File: client/src/components/Download.js [hunk 1]
```diff
@@ -45,10 +51,41 @@ export const Download = ({ icon, resource }) => {
...
+    setToggleFile(true)
+    setInital(false)
+    setDownload(false)
+  }
+
+  useEffect(() => {
+    if (initial || (!initial && !toggleFile)) setResource(initialResource)
+
+    setPublicComputedFile(
+      mutltipleComputedFiles ? null : resource.computed_files[0]
```
Comment (davidsmejia) [inline client/src/components/Download.js:67]: lets write a helper her to get `getDefaultComputedFile` which returns resource.computed_files[0]

File: client/src/components/Download.js [hunk 2]
```diff
@@ -105,10 +142,18 @@ export const Download = ({ icon, resource }) => {
               resource={resource}
               computedFile={download}
               handleSelectFile={handleSelectFile}
+              hasToggleFile={
```
[the remark this answers is not in the document]
Comment (nozomione) [reply client/src/components/Download.js]: Done, thank you!

**Comment by davidsmejia** [comment]
This is looking good. One recommendation would be to instead of assigning `isVisible` on every column. We should just loop over the columns and if that property is not defined, set it to true, since that is the default behavior that we want.

**Comment by nozomione** [comment]
> assigning `isVisible` on every column

Completed, thank you for the feedback!

File: client/src/components/Download.js [hunk 3]
```diff
@@ -1,31 +1,41 @@
...
+  const [togglePublicComputedFile, setTogglePublicComputedFile] =
+    useState(false)
   const [showing, setShowing] = useState(false)
   const [download, setDownload] = useState(false)
   const label = isProjectID(resource.scpca_id)
     ? 'Download Project'
     : 'Download Sample'
 
+  const defaultComputedFile = getDefaultComputedFile(resource)
+  const mutltipleComputedFiles = hasMultiple(resource.computed_files)
```
Comment (davidsmejia) [inline client/src/components/Download.js:37]:
```suggestion
  const multipleComputedFiles = hasMultiple(resource.computed_files)
```

File: client/src/components/WarningText.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import { Box, Paragraph } from 'grommet'
+import { Icon } from 'components/Icon'
+import { Link } from 'components/Link'
+
+export const WarningText = ({
```
Comment (davidsmejia) [inline client/src/components/WarningText.js:6]: Can we use the same interface for passing the margins and pads as grommet? I don't see any benefit of passing in an array and mapping that to the sides.

**Review by davidsmejia — APPROVED** [review]
LGTM just a few suggestions!

File: client/src/components/Download.js (hunk 3 above)
Comment (nozomione) [reply client/src/components/Download.js]: Done, thank you!

File: client/src/components/WarningText.js (hunk 4 above)
Comment (nozomione) [reply client/src/components/WarningText.js:6]: Done, thank you!

**Pull request merged**